velox/functions/prestosql/aggregates/tests/MinMaxByAggregationTest.cpp

```diff
@@ -15,178 +15,1541 @@
  */
 #include <fmt/format.h>
 
+#include "velox/dwio/dwrf/test/utils/BatchMaker.h"
 #include "velox/exec/tests/utils/PlanBuilder.h"
 #include "velox/functions/prestosql/aggregates/AggregateNames.h"
 #include "velox/functions/prestosql/aggregates/tests/AggregationTestBase.h"
 
 using facebook::velox::exec::test::PlanBuilder;
+using namespace facebook::velox::exec::test;
 
 namespace facebook::velox::aggregate::test {
 
 namespace {
 
+using facebook::velox::exec::test::MaterializedRow;
+
 class MinMaxByAggregationTest : public AggregationTestBase {
  protected:
   void testGlobalAggregation(
       const std::vector<RowVectorPtr>& vectors,
       const std::string& aggName,
       const std::string& colName) {
-    std::string funcName = aggName == kMaxBy ? "max" : "min";
-
-    for (int i = 0; i < 6; ++i) {
-      auto valColName = fmt::format("c{}", i);
-
+    const std::string funcName = aggName == kMaxBy ? "max" : "min";
+    for (int i = 0; i < rowType_->size(); ++i) {
+      const auto valColName = fmt::format("c{}", i);
+      const std::string verifyDuckDbSql = "SELECT " + valColName +
+          " FROM tmp WHERE " + colName + " = ( SELECT " + funcName + "(" +
+          colName + ") FROM tmp) LIMIT 1";
+      SCOPED_TRACE(fmt::format(
+          "{}({}, {})\nverifyDuckDbSql: {}",
+          aggName,
+          valColName,
+          colName,
+          verifyDuckDbSql));
       auto op =
           PlanBuilder()
               .values(vectors)
               .partialAggregation(
                   {}, {fmt::format("{}({}, {})", aggName, valColName, colName)})
               .finalAggregation()
               .planNode();
+      assertQuery(op, verifyDuckDbSql);
+    }
+  }
 
-      assertQuery(
-          op,
-          "SELECT " + valColName + " FROM tmp WHERE " + colName +
-              " = ( SELECT " + funcName + "(" + colName +
-              ") FROM tmp) LIMIT 1");
+  void testGroupByAggregation(
+      const std::vector<RowVectorPtr>& vectors,
+      const std::string& aggName,
+      const std::string& colName) {
+    const std::string funcName = aggName == kMaxBy ? "max" : "min";
+    for (int volColIdx = 0; volColIdx < rowType_->size(); ++volColIdx) {
+      const int groupByColIdx = folly::Random::rand32(0, rowType_->size());
```

Comment:
Why random? It will make failures hard to reproduce? Why can't we just test all possible combinations?